diff --git a/soem_beckhoff_drivers/CMakeLists.txt b/soem_beckhoff_drivers/CMakeLists.txt
index 23a5085..1f92c3c 100644
--- a/soem_beckhoff_drivers/CMakeLists.txt
+++ b/soem_beckhoff_drivers/CMakeLists.txt
@@ -32,7 +32,7 @@ find_package(Orocos-RTT REQUIRED ${RTT_HINTS})
 # documentation.
 include(${OROCOS-RTT_USE_FILE_PATH}/UseOROCOS-RTT.cmake)
 
-rosbuild_include(rtt_ros_integration GenerateRTTtypekit)
+rosbuild_include(rtt_rosnode GenerateRTTtypekit)
 
 ros_generate_rtt_typekit(soem_beckhoff_drivers)
 
diff --git a/soem_beckhoff_drivers/manifest.xml b/soem_beckhoff_drivers/manifest.xml
index a1a1cff..506e936 100644
--- a/soem_beckhoff_drivers/manifest.xml
+++ b/soem_beckhoff_drivers/manifest.xml
@@ -11,7 +11,7 @@
   <depend package="soem_core"/>
   <depend package="soem_master"/>
   <depend package="rtt"/>
-  <depend package="rtt_ros_integration"/>
+  <depend package="rtt_rosnode"/>
 
 
 </package>
diff --git a/soem_ebox/CMakeLists.txt b/soem_ebox/CMakeLists.txt
index c9eb799..ca3f811 100644
--- a/soem_ebox/CMakeLists.txt
+++ b/soem_ebox/CMakeLists.txt
@@ -33,7 +33,7 @@ include(${OROCOS-RTT_USE_FILE_PATH}/UseOROCOS-RTT.cmake)
 #uncomment if you have defined messages
 rosbuild_genmsg()
 
-rosbuild_include(rtt_ros_integration GenerateRTTtypekit)
+rosbuild_include(rtt_rosnode GenerateRTTtypekit)
 ros_generate_rtt_typekit(soem_ebox)
 
 #
diff --git a/soem_ebox/manifest.xml b/soem_ebox/manifest.xml
index 3b9ce3e..da3513f 100644
--- a/soem_ebox/manifest.xml
+++ b/soem_ebox/manifest.xml
@@ -5,7 +5,7 @@
     <author>Ruben Smits</author>
     <license>BSD</license>
     <depend package="rtt" />
-    <depend package="rtt_ros_integration" />
+    <depend package="rtt_rosnode" />
     <depend package="soem_core" />
     <depend package="soem_master" />
 </package>
