diff -rupN soem_core/build/soem_core/src/ethercatconfig.c soem_core_bak/build/soem_core/src/ethercatconfig.c
--- soem_core/build/soem_core/src/ethercatconfig.c	2011-06-13 07:40:36.000000000 -0700
+++ soem_core_bak/build/soem_core/src/ethercatconfig.c	2012-09-06 17:09:14.884076970 -0700
@@ -59,7 +59,7 @@
 #include "ethercatconfig.h"
 
 // define if debug printf is needed
-//#define EC_DEBUG
+#define EC_DEBUG
 
 #ifdef EC_DEBUG
 #define EC_PRINT printf
diff -rupN soem_core/build/soem_core/src/nicdrv.c soem_core_bak/build/soem_core/src/nicdrv.c
--- soem_core/build/soem_core/src/nicdrv.c	2012-09-06 17:09:37.230075414 -0700
+++ soem_core_bak/build/soem_core/src/nicdrv.c	2012-09-06 17:09:14.884076970 -0700
@@ -574,7 +574,11 @@ static int ec_waitinframe_red(uint8 idx,
 			if (wkc2 <= EC_NOFRAME)
 				wkc2 = ec_inframe(idx, 1);
 		}	
-		gettimeofday(&tv1, 0);
+		//gettimeofday(&tv1, 0);
+		struct timespec ts1;
+		clock_gettime(CLOCK_REALTIME, &ts1);
+		tv1.tv_sec  = ts1.tv_sec;
+		tv1.tv_usec = ts1.tv_nsec / 1000;
 	/* wait for both frames to arrive or timeout */	
 	} while (((wkc <= EC_NOFRAME) || (wkc2 <= EC_NOFRAME)) && (timercmp(&tv1, &tvs, <)));
 	/* only do redundant functions when in redundant mode */
@@ -608,7 +612,8 @@ static int ec_waitinframe_red(uint8 idx,
 				/* copy primary rx to tx buffer */
 				memcpy(&ec_txbuf[idx][ETH_HEADERSIZE], &ec_rxbuf[idx], ec_txbuflength[idx] - ETH_HEADERSIZE);
 			}
-			gettimeofday(&tv1, 0);				
+			printf("Error! File: %s, was once line 615\n", __FILE__);
+			gettimeofday(&tv1, 0);
 			tv2.tv_sec = 0;
 			tv2.tv_usec = EC_TIMEOUTRET;
 			timeradd(&tv1, &tv2, &tvs);
@@ -644,7 +649,12 @@ int ec_waitinframe(uint8 idx, int timeou
 	int wkc;
 	struct timeval tv1, tv2, tve;
 	
-	gettimeofday(&tv1, 0);
+	//gettimeofday(&tv1, 0);
+	struct timespec ts1;
+	clock_gettime(CLOCK_REALTIME, &ts1);
+	tv1.tv_sec  = ts1.tv_sec;
+	tv1.tv_usec = ts1.tv_nsec / 1000;
+
 	tv2.tv_sec = 0;
 	tv2.tv_usec = timeout;
 	timeradd(&tv1, &tv2, &tve);
